### PR TITLE
less_simple_linear_interpolation can be replaced by np.interp.

### PR DESCRIPTION
**Pull request by anntzer**
Also, in contour (the only place that used it), -1 (instead of nan) is
a perfectly reasonable marker for out of bounds which avoids having to
dance with the errstate.

Replaces #9859: @efiring pointed out and I confirmed that we neither implement nor need linear extrapolation, so np.interp is perfectly sufficient.

Made a separate PR as the approach is somewhat different.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
Tiny image error on the mac: I don't have time right now, but could try to see what the difference is later tonight.

**Comment by anntzer** [comment]
that'd be very helpful, thanks

**Comment by jklymak** [comment]
The image error is:  
```
E           matplotlib.testing.exceptions.ImageComparisonFailure: images not close (RMS 0.015):
E           	result_images/test_patheffects/collection.png
E           	result_images/test_patheffects/collection-expected.png
```

If I look at the diff image, I can't see where the difference is, and the images look fine.  I'd suggest raising the tolerance for this one.

**Comment by jklymak** [comment]
Not sure if github keeps the resolution:

collection-failed-diff

**Comment by anntzer** [comment]
We're probably running into some 32-vs-64 bit rounding error that's a bit bigger for interp; that test actually already had a nonzero tolerance so I feel comfortable increasing it a bit.

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -289,6 +289,7 @@ def get_label_width(self, lev, fmt, fsize):
 
         return lw
 
+    @cbook.deprecated("2.2")
```
Comment (tacaswell) [inline lib/matplotlib/contour.py]: This seems unrelated.  Can you remove it or add an API changes doc?

File: lib/matplotlib/mlab.py [hunk 2]
```diff
@@ -3404,6 +3404,7 @@ def griddata(x, y, z, xi, yi, interp='nn'):
 ##################################################
 # Linear interpolation algorithms
 ##################################################
+@cbook.deprecated("2.2", alternative="np.interp")
```
Comment (tacaswell) [inline lib/matplotlib/mlab.py]: This needs API changes docs.

File: lib/matplotlib/contour.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/contour.py]: moved to its own PR

File: lib/matplotlib/mlab.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/mlab.py]: done

**Comment by jklymak** [comment]
Thanks @anntzer
